components/layout_2020/style_ext.rs

```diff
@@ -466,6 +467,15 @@ impl ComputedValuesExt for ComputedValues {
     /// Returns true if this style has a transform, or perspective property set and
     /// it applies to this element.
     fn has_transform_or_perspective(&self, fragment_flags: FragmentFlags) -> bool {
+        if self.is_transformable(fragment_flags) {
+            !self.get_box().transform.0.is_empty() ||
+                self.get_box().perspective != Perspective::None
+        } else {
+            false
+        }
```

Comment:
Could just be

```suggestion
        self.is_transformable(fragment_flags) &&
            (!self.get_box().transform.0.is_empty() ||
                self.get_box().perspective != Perspective::None)
```

But you can just wait for #35808 to land and then rebase.